### PR TITLE
Correct the order of side-effects in lifted | and &

### LINKED ISSUE #7091: Regression in lifted binary operators evaluation
Roslyn does not follow previous C# compilers evaluation order for lifted binary operators and it's quite counter intuitive.

```csharp
using System;

class MainClass
{
	public static int Main ()
	{
		var res = XX() | YY();		
		return 0;
	}

    static bool XX()
    {
        Console.WriteLine ("XX");
        return true;
    }	

    static bool? YY()
    {
        Console.WriteLine ("YY");
        return true;
    }	
}
```

roslyn output
`
YY
XX
`

previous versions
`
XX
YY
`

### PR DESCRIPTION
**Pull request by gafter**
Fixes #7091 

@dotnet/roslyn-compiler Please review this tiny PR.

### CODE REVIEW

**Comment by AdamSpeight2008** [comment]
Is this also applicable to VB?

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOperators.cs [hunk 1]
```diff
@@ -3310,6 +3310,50 @@ .maxstack  2
             comp.VerifyIL("Program.M", il);
         }
 
+        [WorkItem(7091, "https://github.com/dotnet/roslyn/issues/7091")]
+        [Fact]
+        public void LiftedBitwiseOr()
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOperators.cs]: Does this pattern apply to nullable types as well?  If so I think we should add some tests for that.

Comment (gafter) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOperators.cs]: That is what this test does.

**Comment by AlekseyTs** [comment]
LGTM, assuming VB doesn't have this issue, which should be confirmed.

**Comment by gafter** [comment]
I couldn't find this error in the VB lowering phase, and it appears to do the right thing when tested by hand.

**Comment by TyOverby** [comment]
LGTM

**Comment by VSadov** [comment]
LGTM

**Pull request merged**